pkg/envd/docker.go

```diff
@@ -459,6 +465,30 @@ func (e dockerEngine) StartEnvd(ctx context.Context, so StartOptions) (*StartRes
 	if so.ShmSize > 0 {
 		hostConfig.ShmSize = int64(so.ShmSize) * 1024 * 1024
 	}
+	// resource
+	if len(so.NumCPU) > 0 {
+		cpu, err := strconv.ParseFloat(so.NumCPU, 64)
+		if err != nil {
+			logger.Infof("parse `cpu` error: %v, ignore this argument", err)
+		} else if runtime.GOOS == "windows" {
+			hostConfig.NanoCPUs = int64(cpu * 10e9)
+		} else {
+			// refer to https://docs.docker.com/config/containers/resource_constraints/#configure-the-default-cfs-scheduler
+			// CPU quota and CPU period only work for UNIX platform
+			hostConfig.CPUQuota = int64(cpu * 10e5)
+		}
+	}
+	if len(so.CPUSet) > 0 {
+		hostConfig.CpusetCpus = so.CPUSet
```

Comment:
the cpuset argument and numcpu should be exited one at the same time.

Maybe we can do something like this

1. use a boolean flag named `use-cpu-set` or something else.
2. if the people set the `use-cpu-set` , we can do something below
    1. if the `cpu-set` is not set, we can calculate the CPU core we want to binding.
        1. if `cpu` is a float, we can make it as a integer, such as 0.5 -> 1, 1.5 -> 2. And we can raise a warning during the code execution
        2. if `cpu` is a integer, we can use it directly. such as 1 cpu binding to 0, 2 cpu binding to 0,1 etc.
    2. if the `cpu-set` is set, we can use it directly(`cpu` should be ignored and warning it during in execution)
3. if the people don't set the `use-cpu-set`, we can use the `cpu` directly(raise a warning message if people set `cpu-set`).

Reply:
cc @gaocegege

Reply:
SGTM!

Reply:
I think converting `cpu` to `cpu-set` when `use-cpu-set` is enabled is tricky:
1. round the CPU quota may not be something expected
2. bind to cpu-set requires storing the previous environment information, otherwise we don't know which one is still free to use
3. this may only work well when the server is fully used for envd environments

Thus I think we can just panic if both of them are set.